cmd/wallet-web/src/components/DIDManagement/CreateOrb.vue

```diff
@@ -0,0 +1,132 @@
+        <option value="ECDSAP256IEEEP1363">P-256</option>
+        <option value="ECDSAP384IEEEP1363">P-384</option>
+        <option value="BLS12381G2">BLS12381G2</option>
+      </select>
+    </div>
+
+    <div class="flex flex-col flex-grow justify-start items-start w-full">
+      <label for="select-signature-suite" class="mb-1 text-base">Signature Suite:</label>
+      <select
+        id="select-signature-suite"
+        v-model="signType"
+        class="mb-5 w-full max-w-full border-b"
+      >
+        <option value="Ed25519VerificationKey2018">Ed25519VerificationKey2018</option>
+        <option value="JwsVerificationKey2020">JwsVerificationKey2020</option>
+        <option value="Bls12381G2Key2020">Bls12381G2Key2020</option>
+      </select>
+    </div>
+
+    <div class="flex flex-col flex-grow justify-start items-start w-full">
+      <label for="select-key-purpose" class="mb-1 text-base">Key Purpose:</label>
+      <select id="select-key-purpose" v-model="purpose" class="mb-5 w-full max-w-full border-b">
+        <option value="all">all</option>
+        <option value="authentication">authentication</option>
+        <option value="assertionMethod">assertionMethod</option>
+      </select>
+    </div>
+
+    <button id="createDIDBtn" class="mt-5 btn-primary" @click="createDID">Create and Save</button>
+
+    <div v-if="errors.length">
+      <b>Please correct the following error(s):</b>
+      <ul>
+        <li v-for="error in errors" :key="error">{{ error }}</li>
+      </ul>
+    </div>
+    <div v-if="loading" class="mt-2 ml-4 h-48">
+      <div class="">
+        <spinner />
+      </div>
+    </div>
+    <div v-if="createDIDSuccess">
+      <div class="" style="color: green">
+        <label id="create-did-success" class="text-sm">Saved your DID successfully.</label>
+      </div>
+    </div>
+    <div>
+      <span class="break-words max-w-full">{{ didDocTextArea }}</span>
+    </div>
+  </div>
+</template>
+
+<script>
+import { DIDManager } from '@trustbloc/wallet-sdk';
+import { mapActions, mapGetters } from 'vuex';
+import Spinner from '@/components/Spinner/Spinner';
+
+export default {
+  name: 'CreateOrb',
+  components: {
+    Spinner,
+  },
+  props: {
+    allDIDs: {
+      type: Array,
+      required: true,
+    },
+  },
+  data() {
+    return {
+      didDocTextArea: '',
+      purpose: 'all',
+      keyType: 'ED25519',
+      signType: 'Ed25519VerificationKey2018',
+      errors: [],
+      loading: false,
+      createDIDSuccess: false,
+    };
+  },
+  created: async function () {
+    const agent = this.getAgentInstance();
+    const { user } = this.getCurrentUser().profile;
+    this.didManager = new DIDManager({ agent, user });
+    await Promise.all([this.listDIDs()]);
+  },
+  methods: {
+    ...mapGetters('agent', { getAgentInstance: 'getInstance' }),
+    ...mapGetters(['getCurrentUser', 'getAgentOpts']),
+    ...mapActions(['refreshUserPreference']),
+    listDIDs: async function () {
```

Comment:
just curious, why we are listing DIDs in this component?

Reply:
In order to update the list of all DIDs, which is displayed in the preferences tab, right away, without reload @sudeshrshetty